specs/014-kratos-authid/spec.md

```diff
@@ -0,0 +1,132 @@
+
+---
+
+### Edge Cases
+
+- How does the migration handle users whose Kratos identity no longer exists or cannot be retrieved at runtime?
+- What happens when multiple Kratos identities appear to match the same user (duplicate or merged accounts)?
+- How does the system respond if storing the Kratos identifier for a new user fails due to transient service outages?
+- How should the endpoint behave if Kratos returns partial data (e.g., missing email) required for user creation?
+- Endpoint is accessible only within the internal deployment network; no per-request authentication or allowlisting is required.
+- What recovery steps are needed if the migration fails partway through the existing database migration pipeline?
+
+## Requirements _(mandatory)_
+
+### Functional Requirements
+
+- **FR-001**: The platform MUST persist a Kratos identity identifier for every user record to maintain a permanent linkage between Alkemio and Kratos.
+- **FR-002**: The migration MUST retrieve and populate Kratos identity identifiers for all existing users using the most reliable matching data available today.
+- **FR-003**: The migration MUST record exceptions for any user whose Kratos identifier cannot be resolved so operations can address them manually.
+- **FR-004**: The user onboarding flow MUST store the Kratos identity identifier during account creation before confirming the user is active.
+- **FR-005**: The platform MUST prevent duplicate Kratos identity identifiers from being associated with different user records, ensuring one-to-one mapping.
+- **FR-006**: Administrative interfaces and exports MUST expose the stored Kratos identity identifier to authorized staff for auditing.
+- **FR-007**: The system MUST emit structured logs and metrics when identity synchronization fails so incidents can be triaged promptly.
+- **FR-008**: A private REST endpoint MUST return the Alkemio user ID when provided a valid Kratos identity identifier for an existing user within acceptable performance thresholds.
+- **FR-009**: The private REST endpoint MUST create a new user using Kratos-sourced data when the identity identifier is valid but no user record exists, following the same validation and enrichment rules as first-login flows.
+- **FR-010**: The endpoint MUST be exposed exclusively on the internal deployment network with no per-request authentication requirement; audit logs MUST still capture each resolution or creation event for traceability.
+- **FR-011**: The platform MUST gracefully handle Kratos outages or invalid responses during endpoint calls by applying retries/backoff and returning actionable errors without leaving inconsistent user records.
+- **FR-012**: The migration effort MUST integrate with the current database migration workflow so it can be executed, rolled back, and validated through the established process without introducing parallel tooling.
+
```

Comment:
`FR-013` -- current behavior on removing on of alkemio user / kratos user is preserved ? TBD